axelrod/strategies/cycler.py

```diff
@@ -72,9 +72,11 @@ def __init__(self, cycle="CCD") -> None:
         self.classifier['memory_depth'] = len(cycle) - 1
 
     def strategy(self, opponent: Player) -> Action:
+        actions = {'C': Actions.C, 'D': Actions.D}
         curent_round = len(self.history)
         index = curent_round % len(self.cycle)
-        return self.cycle[index]
+        action_str = self.cycle[index]
+        return actions[action_str]
```

Comment:
This is due to the automatic reset method which is not recognising the cycle as a genorator:

https://github.com/Axelrod-Python/Axelrod/blob/master/axelrod/tests/strategies/test_player.py#L263

```python
>>> import itertools
>>> import types
>>> isinstance((k for k in [1, 2, 3]), types.GeneratorType)
True
>>> isinstance(itertools.cycle([1, 2, 3]), types.GeneratorType)
False
>>> isinstance(itertools.cycle([1, 2, 3]), itertools.cycle)
True

```

I suggest you add another check for cycles at this line: https://github.com/Axelrod-Python/Axelrod/blob/master/axelrod/tests/strategies/test_player.py#L263

```
-  elif isinstance(reset_value, types.GeneratorType) 
+ elif isinstance(reset_value, types.GeneratorType) or isinstance(reset_value, itertools.cycle):
```

That should check those things properly :)